src/NServiceBus.Transport.SQS.TransportTests.DoNotWrapOutgoingMessages/Sending_messages_with_invalid_sqs_chars.cs

```diff
@@ -1,38 +1,47 @@
 ﻿namespace NServiceBus.TransportTests
 {
-    using System;
     using System.Collections.Generic;
     using System.Threading.Tasks;
     using NUnit.Framework;
     using Transport;
 
-    public class Sending_metrics_messages : NServiceBusTransportTest
+    public class Sending_messages_with_invalid_sqs_chars : NServiceBusTransportTest
     {
         [TestCase(TransportTransactionMode.None)]
         [TestCase(TransportTransactionMode.ReceiveOnly)]
-        public async Task Should_not_fail_when_using_do_not_wrap(
+        public async Task Should_receive_message(
             TransportTransactionMode transactionMode)
         {
             var messageProcessed = CreateTaskCompletionSource<MessageContext>();
+            byte[] copyOfTheBody = null;
 
             await StartPump(
-                (context, _) => messageProcessed.SetCompleted(context),
+                (context, _) =>
+                {
+                    // This is crucial due to internal buffer pooling in SQS transport
+                    copyOfTheBody = context.Body.ToArray();
+                    return messageProcessed.SetCompleted(context);
+                },
                 (_, __) => Task.FromResult(ErrorHandleResult.Handled),
                 TransportTransactionMode.None);
 
             var headers = new Dictionary<string, string>
             {
-                { Transport.SQS.Constants.MetricsMessageMetricTypeHeaderKey, "doesn't matter" },
-                { Headers.ContentType, Transport.SQS.Constants.MetricsMessageContentTypeHeaderValue }
+                { "SomeHeader", "doesn't matter" },
```

Comment:
Would it make sense to add some invalid characters here as well to make this more future proof should we switch to using [message attributes](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-message-metadata.html) in the future when the "max 10" constraint have been lifted?